src/audio_processing.rs

```diff
@@ -0,0 +1,111 @@
+use dasp::Sample;
+use dasp_signal::Signal;
+use dasp_window::Window;
+use rustfft::{num_complex::Complex, num_traits::ToPrimitive};
+use std::sync::Arc;
+
+pub struct FftResult {
+    pub raw_bins: Vec<Complex<f64>>,
+    sample_rate: usize,
+    fft_size: usize,
+}
+
+impl FftResult {
+    pub fn new(raw_bins: Vec<Complex<f64>>) -> Self {
+        Self {
+            raw_bins,
+            sample_rate: 48000,
+            fft_size: 1024,
+        }
+    }
+
+    pub fn get_frequency_bin(&self, frequency: usize) -> Option<f64> {
+        let bin_size = (self.sample_rate / 2) / (self.raw_bins.len() / 2);
```

Comment:
could be simplified to: `self.sample_rate / self.raw_bins.len()`, but the real formula is this.